python/tk_multi_workfiles/actions/file_action_factory.py

```diff
@@ -36,158 +37,336 @@
-        change_work_area = (work_area.context != app.context)
-        if change_work_area and in_other_users_sandbox:
-            user_work_area = work_area.create_copy_for_user(g_user_cache.current_user)
-            change_work_area = (user_work_area.context != app.context)
+        self._user_work_area = work_area
+        self._change_work_area = (work_area.context != app.context)
+        if self._change_work_area and self._in_other_users_sandbox:
+            self._user_work_area = work_area.create_copy_for_user(g_user_cache.current_user)
+            self._change_work_area = (self._user_work_area.context != app.context)
 
         # and if it's possible to copy this file to the work area:
-        can_copy_to_work_area = False
-        if change_work_area and app.context:
+        self._can_copy_to_work_area = False
+        if self._change_work_area and app.context:
             current_env = WorkArea(app.context)
-            can_copy_to_work_area = current_env.work_template is not None
+            self._can_copy_to_work_area = current_env.work_template is not None
             # (AD) TODO - it's possible the work template for the current work area has different requirements than
-            # the source work area (e.g. it may require a name where the source work template doesn't!)  This should 
+            # the source work area (e.g. it may require a name where the source work template doesn't!)  This should
             # probably check that file.path is translatable to the current work area (contains all the required keys)
             # in addition to the simple check it's currently doing.
 
-        # get the list of file versions for the file:
-        file_versions = file.versions
+    def get_actions(self, file_item):
+        """
+        Retrives the list of actions for the given file
+
+        :returns: List of Actions.
+        """
+        actions = []
+
+        current_user_file_versions = self._get_current_user_file_versions(file_item)
+
+        # add the interactive 'open' action.  This is the
+        # default/generic open action that gets run whenever someone
+        # double-clicks on a file or just hits the 'Open' button
+        actions.append(
+            InteractiveOpenAction(
+                file_item,
+                current_user_file_versions,
+                self._work_area,
+                self._workfiles_visible,
+                self._publishes_visible
+            )
+        )
+
+        actions.extend(self._create_local_file_actions(file_item, current_user_file_versions))
+        actions.extend(self._create_published_file_actions(file_item, current_user_file_versions))
+        actions.extend(self._create_previous_versions_actions_menus(file_item))
+        actions.extend(self._create_new_file_action())
+        actions.extend(self._create_custom_actions(file_item, current_user_file_versions))
+        actions.extend(self._create_show_in_actions(file_item, current_user_file_versions))
```

Comment:
wondering if this would be a good place to just give a quick overview of what each of the groups of actions do or what they're used for. maybe with examples. not really necessary, just might help future devs when processing the code.